P/PETSc/build_tarballs.jl

```diff
@@ -99,15 +99,21 @@ build_petsc double real Int64
 build_petsc single real Int64
 build_petsc double complex Int64
 build_petsc single complex Int64
+
+# On windows move back since we can't change the install name
+if [[ "${target}" == *-mingw32* ]]; then
+   mv ${libdir}/petsc/double_real_Int32/lib/libpetsc_double_real_Int32.${dlext} "${libdir}/petsc/double_real_Int32/lib/libpetsc.${dlext}.3.6.5" # TODO fix version suffix
```

Comment:
The problem I have with this is: if we do the same for all variants, how do we distinguish between all of them if they have all the same name?

Reply:
Apparently Windows cares about the path being loaded from and not just the so name.
But yes this was just aimed at unblocking one particular config.

Reply:
If we have all the libraries called libpetsc.dll and we have a downstream library linking to libpetsc.dll, which one will actually link to?

Reply:
That's anyone's guess. Right now we solve it by having no library be able to link against PETSc on windows xD

Reply:
> That's anyone's guess.

Eh, that's my problem with PETSc 🙂 

> Right now we solve it by having no library be able to link against PETSc on windows xD

Sure, I understand your concern, it's just that I believe this is really something that should be fixed upstream.